lib/mysql.js

```diff
@@ -345,18 +348,23 @@ MySQL.prototype.toDatabase = function (prop, val, forCreate) {
     return val ? 'Point(' + val.lat + ',' + val.lng + ')' : 'NULL';
   }
   if (prop.type === Object) {
-    return this.client.escape(val);
+    return serializeObject(val);
   }
   if (typeof prop.type === 'function') {
-    if (prop.type.modelName) {
-      // For embedded models
-      return this.client.escape(JSON.stringify(val));
-    }
-    return this.client.escape(prop.type(val));
+    return serializeObject(val);
   }
-  return this.client.escape(val.toString());
+  return serializeObject(val);
 };
 
+function serializeObject(obj) {
+  var str = JSON.stringify(obj);
+  if (str && str[0] === '"' && str[str.length - 1] === '"') {
+    // Remove the " for string values
+    str = str.substring(1, str.length - 1);
+  }
+  return "'" + str + "'";
```

Comment:
This creates a SQL injection vulnerability because `JSON.stringify` is not escaping single quotes.

```
// io.js 1.4.2
> console.log(JSON.stringify("a'b"))
"a'b"
```